applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_law.cpp

```diff
@@ -0,0 +1,202 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Anne van de Graaf
+//
+
+#include "linear_elastic_law.h"
+#include "includes/mat_variables.h"
+
+namespace Kratos
+{
+
+bool GeoLinearElasticLaw::RequiresInitializeMaterialResponse()
+{
+    // Avoid that `ConstitutiveLaw::InitializeMaterialResponse` throws
+    // an exception
+    return false;
+}
+
+bool GeoLinearElasticLaw::RequiresFinalizeMaterialResponse()
+{
+    // Avoid that `ConstitutiveLaw::FinalizeMaterialResponse` throws
+    // an exception
+    return false;
+}
+
+ConstitutiveLaw::StrainMeasure GeoLinearElasticLaw::GetStrainMeasure()
+{
+    return StrainMeasure_Infinitesimal;
+}
+
+ConstitutiveLaw::StressMeasure GeoLinearElasticLaw::GetStressMeasure()
+{
+    return StressMeasure_Cauchy;
+}
+
+// NOTE: Since we are in the hypothesis of small strains we can use the same function for everything
+void GeoLinearElasticLaw::CalculateMaterialResponsePK1(ConstitutiveLaw::Parameters& rValues)
+{
+    CalculateMaterialResponsePK2(rValues);
+}
+
+void GeoLinearElasticLaw::CalculateMaterialResponsePK2(ConstitutiveLaw::Parameters& rValues)
+{
+    KRATOS_TRY
+    // b.- Get Values to compute the constitutive law:
+    const Flags& r_options = rValues.GetOptions();
+
+    Vector& r_strain_vector = rValues.GetStrainVector();
+
+    // NOTE: SINCE THE ELEMENT IS IN SMALL STRAINS WE CAN USE ANY STRAIN MEASURE. HERE EMPLOYING THE CAUCHY_GREEN
+    if (r_options.IsNot(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN)) {
+        CalculateCauchyGreenStrain(rValues, r_strain_vector);
+    }
+
+    if (r_options.Is(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR)) {
+        Matrix& r_constitutive_matrix = rValues.GetConstitutiveMatrix();
+        CalculateElasticMatrix(r_constitutive_matrix, rValues);
+    }
+
+    if (r_options.Is(ConstitutiveLaw::COMPUTE_STRESS)) {
+        Vector& r_stress_vector = rValues.GetStressVector();
+        CalculatePK2Stress(r_strain_vector, r_stress_vector, rValues);
+    }
+
+    KRATOS_CATCH("")
+}
+
+void GeoLinearElasticLaw::CalculateMaterialResponseKirchhoff(ConstitutiveLaw::Parameters& rValues)
+{
+    CalculateMaterialResponsePK2(rValues);
```

Comment:
A general question, most likely for a separate issue, here CalculateMaterialResponseKirchhoff calls CalculateMaterialResponsePK2. Below CalculateMaterialResponseCauchy calls CalculateMaterialResponsePK2. When SmallStrainUDSM3DLaw::CalculateMaterialResponsePK2 calls CalculateMaterialResponseCauchy. It looks that in both cases function names do not reflect the numerics and here CalculateMaterialResponsePK2 should be split in appropriate functions.